saige_assoc.py

```diff
@@ -0,0 +1,300 @@
+    """
+
+
+# Combine single variant associations at gene level (step 3)
+def build_obtain_gene_level_pvals_command(
+    gene_name: str,
+    saige_sv_output_file: str,
+    saige_gene_pval_output_file: str,
+):
+    """Build SAIGE command to obtain gene-level pvals
+    Only for single-variant tests (Step3)
+    combines single-variant p-values to obtain one gene
+    level p-value
+
+    Input:
+    - ouput of previous step, association file (txt)
+    - gene we need to aggregate results for (across SNPs)
+    - path for output file
+    """
+    saige_command_step3 = '/usr/local/bin/Rscript step3_gene_pvalue_qtl.R'
+    saige_command_step3 += f' --assocFile={saige_sv_output_file}'
+    saige_command_step3 += f' --geneName={gene_name}'
+    saige_command_step3 += f' --genePval_outputFile={saige_gene_pval_output_file}'
+    return saige_command_step3
+
+
+config = get_config()
+
+
+@click.command()
+@click.option(
+    '--pheno-cov-filename',
+    default='/usr/local/bin/seed_1_100_nindep_100_ncell_100_lambda_2_tauIntraSample_0.5_Poisson.txt',
+)
+@click.option('--vcf-file-path', default='/usr/local/bin/genotype_10markers.vcf.gz')
+@click.option('--covs-list', default='X1,X2,pf1,pf2')
+@click.option('--sample-covs-list', default='X1,X2')
+@click.option('--sample-id', default='IND_ID')
+@click.option(
+    '--null-output-path',
+    default=output_path('nindep_100_ncell_100_lambda_2_tauIntraSample_0.5_gene_1'),
+)
+@click.option(
+    '--sv-output-path',
+    default=output_path('nindep_100_ncell_100_lambda_2_tauIntraSample_0.5_gene_1_cis'),
+)
+@click.option(
+    '--gene-pvals-output-path',
+    default=output_path(
+        'nindep_100_ncell_100_lambda_2_tauIntraSample_0.5_gene_1_cis_genePval'
+    ),
+)
+@click.option('--plink-path', default='/usr/local/bin/n.indep_100_n.cell_1_01.step1')
+@click.option('--gene-name', default='gene_1')
+@click.option('--chrom', default='2')
+@click.option('--cis-window-file', default='/usr/local/bin/gene_1_cis_region.txt')
+@click.option('--fit-null-mem', default='10Gi')
+@click.option('--run-sv-assoc-mem', default='10Gi')
+@click.option('--get-gene-pvals-mem', default='10Gi')
+def association_pipeline(
+    pheno_cov_filename: str,
+    vcf_file_path: str,
+    covs_list: str,
+    sample_covs_list: str,
+    sample_id: str,
+    null_output_path: str,
+    sv_output_path: str,
+    gene_pvals_output_path: str,
+    plink_path: str,
+    gene_name: str,
+    chrom: str,
+    cis_window_file: str,
+    fit_null_mem: str,
+    run_sv_assoc_mem: str,
+    get_gene_pvals_mem: str,
+):
+    """
+    Run association for one gene
+    """
+
+    batch = get_batch('SAIGE-QTL pipeline')
+
+    # step 1 (fit null)
+    fit_null_job = batch.new_job(name='fit null')
+    fit_null_job.image(SAIGE_QTL_IMAGE)
+    fit_null_job.storage(fit_null_mem)
+    fit_null_job.declare_resource_group(
+        output={
+            'rda': f'{root}.rda',
+            'varianceRatio': f'{root}.varianceRatio.txt',
```

Comment:
```suggestion
            'rda': '{root}.rda',
            'varianceRatio': '{root}.varianceRatio.txt',
```

Matt is bad at life, sorry @annacuomo. Reference [here](https://hail.is/docs/batch/api/batch/hailtop.batch.job.BashJob.html#hailtop.batch.job.BashJob.declare_resource_group)